addon/components/file-browser/component.js

```diff
@@ -373,6 +398,107 @@ export default Ember.Component.extend(Analytics, {
         },
         dismissPop() {
             this.set('popupOpen', false);
-        }
+        },
+        checkNodeTitleKeypress(value) {
+            this.set('nodeTitle', value);
+            if (this.get('nodeTitle')) {
+                this.set('isInputInvalid', false);
+            } else {
+                this.set('isInputInvalid', true);
+            }
+        },
+        changeProjectSelectState(state) {
+            this.set('projectSelectState', state);
+            if (this.get('convertOrCopy')) {
+                this.set('convertOrCopy', null);
+            }
+            if (!this.get('isInputInvalid')) {
+                this.set('isInputInvalid', true);
+            }
+        },
+        updateConvertOrCopy(state) {
+            this.set('convertOrCopy', state);
+            if (this.get('convertOrCopy') == 'convert') {
+                this.set('isInputInvalid', false);
+            } else {
+                this.set('isInputInvalid', true);
+            }
+        },
+        setSelectedNode(node) {
+            this.set('node', node);
+            this.set('nodeLink', node.get('links.html'));
+        },
+        setMoveFile() {
+            if (this.get('projectSelectState') == 'newProject') {
+                this.send('createProject');
+            } else if (this.get('projectSelectState') == 'existingProject') {
+                if (this.get('convertOrCopy') == 'copy') {
+                    this.send('createComponent');
+                } else if (this.get('convertOrCopy') == 'convert') {
+                    this.send('moveToComponent');
+                }
+            }
+        },
+        createProject() {
+            let item = this.get('selectedItems.firstObject');
+
+            this.get('store').createRecord('node', {
+                public: false,
+                category: 'project',
+                title: this.get('nodeTitle'),
+            }).save()
+                .then(node => {
+                    this.set('node', node);
+                    this.set('newNode', true);
+                    this.set('nodeLink', node.get('links.html'));
+                    this.set('newProject', true);
+                    this.set('newComponent', false);
+                    this.send('moveFile', item, node);
+                })
+                .catch(() => {
+                    this.get('toast').error(this.get('i18n').t('eosf.components.moveToProject.couldNotCreateProject'));
+                });
+        },
+        createComponent() {
+            let item = this.get('selectedItems.firstObject');
+            let node = this.get('node');
+            node
+                .addChild(this.get('nodeTitle'), null, null, true)
+                .then(child => {
+                    this.set('node', child);
+                    this.set('nodeLink', this.get('node.links.html'));
```

Comment:
Perhaps this should be a computed alias? Or just use `node.links.html` directly?